met-api/src/met_api/models/submission.py

```diff
@@ -52,6 +52,8 @@ def create(cls, submission: SubmissionSchema, session=None) -> DefaultMethodResu
             db.session.commit()
         else:
             session.add(new_submission)
+            session.flush()
+            session.refresh(new_submission)
```

Comment:
I never had to refresh explicitly , whats the reason here?

Reply:
because its part of the transaction for some reason the id does not get reflected in the model object until a flush and refresh, that said i haven't tested with the flush only

Reply:
refresh removed since flush is enough to return its id!

Reply:
yeah..for id , flush does the job